### PR TITLE
follow up #17612

### PR DESCRIPTION
**Pull request by ringabout**

### CODE REVIEW

File: lib/pure/os.nim [hunk 1]
```diff
@@ -2283,9 +2283,9 @@ iterator walkDir*(dir: string; relative = false, checkDir = false):
     for k, v in items(staticWalkDir(dir, relative)):
       yield (k, v)
   else:
-    when weirdTarget:
-      for k, v in items(staticWalkDir(dir, relative)):
-        yield (k, v)
+    when defined(nimscript): discard # taken care of by nimvm but this branch is needed
+    elif defined(js):
+      {.error: "walkDir is not implemented for JS backend".}
```
Comment (ringabout) [inline lib/pure/os.nim]: yeah, it needs to be deferred to runtime(use doAssert)

Comment (ringabout) [reply lib/pure/os.nim]: wait for https://github.com/nim-lang/Nim/pull/17649

Comment (ringabout) [inline lib/pure/os.nim]:
```suggestion
      doAssert false, "walkDir is not implemented for JS backend"
```

Comment (ringabout) [reply lib/pure/os.nim]: no need to wait any more

**Review by timotheecour — APPROVED** [review]

**Pull request closed without merging**